conda/cli/main_run.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+from os.path import isfile, join
+from subprocess import Popen
+import sys
+from tempfile import NamedTemporaryFile
+
+from .. import CondaError
+from ..base.context import context
+from ..common.compat import ensure_binary, iteritems, on_win
+from ..exceptions import CommandNotFoundError
+from ..gateways.disk.delete import rm_rf
+from ..gateways.subprocess import subprocess_call
+
+
+class ExecutableNotFound(CondaError):
+
+    def __init__(self, target_prefix, executable_name):
+        message = ("The executable was not found in the target prefix.\n"
+                   "  target prefix: %(target_prefix)s\n"
+                   "  executable name: %(executable_name)s"
+                   )
+        super(ExecutableNotFound, self).__init__(message, target_prefix=target_prefix,
+                                                 executable_name=executable_name)
+
+
+def _get_activated_env_vars_win(env_location):
+    try:
+        conda_bat = os.environ["CONDA_BAT"]
+    except KeyError:
+        raise CommandNotFoundError("run")
+
+    temp_path = None
+    try:
+        with NamedTemporaryFile('w+b', suffix='.bat', delete=False) as tf:
+            temp_path = tf.name
+            tf.write(ensure_binary(
+                "@%CONDA_PYTHON_EXE% -c \"import os, json; print(json.dumps(dict(os.environ)))\""
+            ))
+        cmd_builder = [
+            "cmd.exe /K \"",
+            "@SET PROMPT= ",
+            "&&",
+            "@SET CONDA_CHANGEPS1=false"
+            "&&",
+            "@CALL {0} activate \"{1}\"".format(conda_bat, env_location),
+            "&&",
+            "\"{0}\"".format(tf.name),
```

Comment:
Why can't we put the user's requested program _right here_, or perhaps inside the temporary script you're creating above?